Sources/AsyncAlgorithms/AsyncZipLatestFromSequence.swift

```diff
@@ -0,0 +1,382 @@
+//
+//  AsyncZipLatestSequence.swift
+//  
+//
+//  Created by Thibault Wittemberg on 31/03/2022.
+//
+
+public extension AsyncSequence {
+  /// Combines `self` with another ``AsyncSequence`` into a single ``AsyncSequence`` where each
+  /// element from `self` is aggregated to the latest known element from the `other` sequence (if any) as a tuple.
+  ///
+  /// Remark: as the `other` sequence is being iterated over in the context of its own ``Task``, there is no guarantee
+  /// that its latest know element is the one that has just been produced when the base sequence produces its next element.
+  ///
+  /// ```
+  /// let base = AsyncChannel<Int>()
+  /// let other = AsyncChannel<String>()
+  /// let sequence = base.zipLatest(from: other)
+  ///
+  /// Task {
+  ///   for element in await sequence {
+  ///    print(element)
+  ///   }
+  /// }
+  ///
+  /// await other.send("a")
+  /// await other.send("b")
+  ///
+  /// ... later in the application flow
+  ///
+  /// await base.send(1)
+  ///
+  /// // will print: (1, "b")
+  /// ```
+  ///
+  /// - Parameter other: the other ``AsyncSequence``
+  /// - Returns: an ``AsyncZipLatestFromSequence`` where elements are a tuple of an element from `self` and the
+  /// latest known element (if any) from the `other` sequence.
+  @inlinable
```

Comment:
I have a feeling that these probably should not be inlinable. Usually the bar I used was "could this reasonably be used for byte streams?" If the answer was yes then it should be inlined/frozen etc. If the answer is meh that doesn't make any sense then they are ok to be resilient boundaries. If the answer is on the fence then it is done if and only if it can be done w/o exposing hacks/workarounds for the language etc.

Reply:
removed :-)